modules/core/src/components/stream-settings-panel.js

```diff
@@ -162,12 +162,13 @@ class StreamSettingsPanel extends PureComponent {
 
   componentWillReceiveProps(nextProps) {
     let {data, values} = this.state;
+    let dataChanged = false;
 
     if (nextProps.streamMetadata !== this.props.streamMetadata) {
       data = createFormData(nextProps.streamMetadata, nextProps);
-      values = null;
+      dataChanged = true;
```

Comment:
Why change the check logic since `log.getStreamMetadata()` is equivalent to the previous `metadata && metadata.streams`? (i.e. change from checking on ` values` VS `dataChanged`)